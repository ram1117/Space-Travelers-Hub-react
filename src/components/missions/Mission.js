import { Badge, Button } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import { bookMission } from '../../redux/missions/missionSlice';

const Mission = ({
  id,
  name,
  description,
  isReserved,
}) => {
  const dispatch = useDispatch();
  return (
    <tr>
      <td>{name}</td>
      <td className="px-5">{description}</td>
      <td className="align-middle">
        <Badge
          bg={isReserved ? 'info' : 'secondary'}
          className="fs-6"
        >
          {isReserved ? 'Active Member' : 'Not A Member'}
        </Badge>
      </td>
      <td className="align-middle">
        <Button
          className="fs-6 mx-3 text-nowrap"
          variant={isReserved ? 'outline-danger' : 'outline-secondary'}
          onClick={() => dispatch(bookMission(id))}
        >
          {isReserved ? 'Leave Mission' : 'Join Mission'}
        </Button>
      </td>
    </tr>
  );
};

Mission.defaultProps = {
  id: '',
  name: '',
  description: '',
  isReserved: false,
};
Mission.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  isReserved: PropTypes.bool,
};

export default Mission;
