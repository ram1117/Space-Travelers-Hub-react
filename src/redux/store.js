import { configureStore } from '@reduxjs/toolkit';
// import logger from 'redux-logger';
import missionReducer from './missions/missionSlice';
import rocketsReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionReducer,
  },
  // middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
