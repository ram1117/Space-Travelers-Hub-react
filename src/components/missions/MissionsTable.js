import { Table } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import Mission from './Mission';
import styles from '../../styles/MissionsTable.module.css';

const MissionsTable = () => {
  const list = useSelector((state) => state.missions.missions);
  return (
    <div>
      <Table
        striped
        bordered
        hover
        responsive="sm"
        className={styles.missionTable}
      >
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {list.map((obj) => (
            <Mission
              key={obj.id}
              id={obj.id}
              name={obj.name}
              description={obj.description}
              isReserved={obj.isReserved}
            />
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default MissionsTable;
