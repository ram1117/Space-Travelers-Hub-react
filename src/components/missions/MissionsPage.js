import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import MissionsTable from './MissionsTable';
import { fetchMissions } from '../../redux/missions/missionSlice';

const MissionsPage = () => {
  const dispatch = useDispatch();
  const status = useSelector((state) => state.missions.toFetch);
  useEffect(() => {
    if (status) {
      dispatch(fetchMissions());
    }
  }, [status, dispatch]);
  return (
      <MissionsTable />
  );
};

export default MissionsPage;
