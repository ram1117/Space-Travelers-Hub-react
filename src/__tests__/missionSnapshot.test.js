import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MissionsTable from '../components/missions/MissionsTable';
import MissionsPage from '../components/missions/MissionsPage';
import Mission from '../components/missions/Mission';

describe('testing snapshots of Mission components', () => {

  it('testing Mission component', () => {
    const mission = renderer.create(<Provider store={store}><Mission /></Provider>).toJSON();
    expect(mission).toMatchSnapshot();
  });
  it('testing MissionsTable component', () => {

    const missionTable = renderer.create(<Provider store={store}><MissionsTable /></Provider>).toJSON();
    expect(missionTable).toMatchSnapshot();
  });
  it('testing MissionsPage component', () => {
    const missionPage = renderer.create(<Provider store={store}><MissionsPage /></Provider>).toJSON();
    expect(missionPage).toMatchSnapshot();
  });
});
