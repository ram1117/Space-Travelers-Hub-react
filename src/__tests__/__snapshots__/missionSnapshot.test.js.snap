// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshots of Mission components testing Mission component 1`] = `
<tr>
  <td />
  <td
    className="px-5"
  />
  <td
    className="align-middle"
  >
    <span
      className="fs-6 badge bg-secondary"
    >
      Not A Member
    </span>
  </td>
  <td
    className="align-middle"
  >
    <button
      className="fs-6 mx-3 text-nowrap btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;

exports[`testing snapshots of Mission components testing MissionsPage component 1`] = `
<div>
  <div
    className="table-responsive-sm"
  >
    <table
      className="missionTable table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`testing snapshots of Mission components testing MissionsTable component 1`] = `
<div>
  <div
    className="table-responsive-sm"
  >
    <table
      className="missionTable table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
